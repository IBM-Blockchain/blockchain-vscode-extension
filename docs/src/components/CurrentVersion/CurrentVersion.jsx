import React, { Component } from 'react';
import Axios from 'axios';

var githubProject = 'IBM-Blockchain/blockchain-vscode-extension';

//Github URLs
var currentReleaseApiUrl = 'https://api.github.com/repos/'+githubProject+'/releases/latest';

class CurrentVersion extends Component {

    constructor() {
        super();
        this.state = {
            currentVersion: undefined
        }
    }

    async componentDidMount(){
        const response = await Axios.get(currentReleaseApiUrl);
        if (response) {
            this.setState({
                currentVersion : response.data.name
            })
        };
    }

    render(){
        if (this.state.currentVersion) {
            return(
                <p>
                    {'Latest version: ' + this.state.currentVersion}
                </p>
            );
        } else {
            return <div/>;
        }
    }
}

export default CurrentVersion;