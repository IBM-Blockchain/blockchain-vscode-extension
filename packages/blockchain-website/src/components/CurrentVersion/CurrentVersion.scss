.latest-version {
    font-size: 14px;
    color: #161616;
    letter-spacing: 0.16px;
    line-height: 18px;
}

.current-version-link {
    text-decoration: none;
    font-size: 14px;
    color: #0F62FE;
    letter-spacing: 0.16px;
    line-height: 18px;
}