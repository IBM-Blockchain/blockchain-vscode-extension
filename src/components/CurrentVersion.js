import React, { Component } from 'react';
import $ from "jquery";


var githubProject = 'IBM-Blockchain/blockchain-vscode-extension';

//Github URLs
var currentReleaseApiUrl = 'https://api.github.com/repos/'+githubProject+'/releases/latest';

class CurrentVersion extends Component {

    constructor() {
        super();
        this.state = {
            currentVersion: undefined,
            error: undefined,
            refreshing: false
        }

        this.refreshVersion = this.refreshVersion.bind(this);
    }

    componentDidMount(){
        this.refreshVersion()
    }

    componentWillUnmount(){
    }

    refreshVersion(e){

        if (e !== undefined) {
            e.preventDefault();
        }

        this.setState({
            refreshing : true
        });

        var latestRelease = function(data){
            console.log(data);
            this.setState({
                currentVersion : data
            });
            //we add a timeout to give some UI feedback on the loading even if it is instantaneous
            setTimeout(function(){
                this.setState({
                    refreshing : false
                })
            }.bind(this),200);
        }.bind(this)

        $.get({
            url : currentReleaseApiUrl,
            success : latestRelease,
            error : function(){
                //we add a timeout to give some UI feedback on the loading even if it is instantaneous
                setTimeout(function(){
                    this.setState({
                        refreshing : false
                    });
                }.bind(this),200);
                this.setState({
                    error : 'Cannot get latest release from GitHub'
                });
            }.bind(this)
        });
    }

    getInitialState(){
        return {
            refreshing : false
        }
    }

    render(){
        if (this.state.currentVersion || this.state.error) {
            return(
                <p>
                    {'Latest version: ' + this.state.currentVersion.name}
                </p>
            );
        } else {
            return <></>;
        }
    }
}

export default CurrentVersion;