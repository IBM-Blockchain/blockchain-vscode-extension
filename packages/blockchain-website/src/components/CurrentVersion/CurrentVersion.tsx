import React, { Component } from 'react';
import Axios, { AxiosResponse } from 'axios';
import './CurrentVersion.scss';

const githubProject: string = 'IBM-Blockchain/blockchain-vscode-extension';

// Github URLs
const currentReleaseApiUrl: string = 'https://api.github.com/repos/' + githubProject + '/releases/latest';
const allReleases: string = 'https://github.com/' + githubProject + '/releases/';

interface VersionState {
    currentVersion: string;
}

class CurrentVersion extends Component<{}, VersionState> {

    constructor(props: Readonly<VersionState>) {
        super(props);
        this.state = {
            currentVersion: ''
        };
    }

    async componentDidMount(): Promise<void> {
        try {
            const response: AxiosResponse = await Axios.get(currentReleaseApiUrl);
            this.setState({
                currentVersion : response.data.name
            });
        } catch (error) {
            return;
        }
    }

    render(): JSX.Element {
        if (this.state.currentVersion.length !== 0) {
            return(
                <p className='latest-version'>
                    {'Latest release: '}
                    <a className='current-version-link' href={allReleases}>
                        {this.state.currentVersion}
                    </a>
                </p>
            );
        } else {
            return <div/>;
        }
    }
}

export default CurrentVersion;
