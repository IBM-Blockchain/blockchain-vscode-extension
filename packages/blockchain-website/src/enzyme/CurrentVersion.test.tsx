import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import Axios from 'axios';
import sinonChai from 'sinon-chai';
import CurrentVersion from '../components/CurrentVersion/CurrentVersion';
chai.should();
chai.use(sinonChai);

const githubProject: string = 'IBM-Blockchain/blockchain-vscode-extension';

// Github URLs
const currentReleaseApiUrl: string = 'https://api.github.com/repos/' + githubProject + '/releases/latest';

describe('CurrentVersion', () => {
    let mySandBox: sinon.SinonSandbox;
    let axiosGetStub: sinon.SinonStub;

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
        axiosGetStub = mySandBox.stub(Axios, 'get');
        axiosGetStub.returns({
            data: {name: 'someVersion'}
        });
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<CurrentVersion />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should output latest version when successful api call', async () => {
        const component: ReactWrapper<{}, {currentVersion: string}, CurrentVersion> = await mount(<CurrentVersion/>);
        axiosGetStub.should.have.been.calledOnceWithExactly(currentReleaseApiUrl);
        component.state().currentVersion.should.equal('someVersion');
    });

    it('should output nothing when unsuccessful api call', async () => {
        axiosGetStub.returns(undefined);
        const component: ReactWrapper<{}, {currentVersion: string}, CurrentVersion> = await mount(<CurrentVersion/>);
        axiosGetStub.should.have.been.calledOnceWithExactly(currentReleaseApiUrl);
        component.state().should.deep.equal({
            currentVersion: ''
        });
    });
});
