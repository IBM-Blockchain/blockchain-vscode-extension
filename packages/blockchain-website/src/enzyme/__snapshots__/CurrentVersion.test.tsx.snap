// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentVersion should render the expected snapshot 1`] = `<div />`;
